server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -91,6 +92,22 @@ public Span getSpan(String tenantId, String id) {
         if (!response.isSourceEmpty()) {
             try {
                 span = deserialize(response.getSourceAsString(), Span.class);
+
+                /**
+                 * Enrich server span with client span annotations
+                 */
+                if (span.serverSpan() && span.url() == null) {
+                    Span clientSpan = getSpan(tenantId, SpanUniqueIdGenerator.getClientId(span.getId()));
+                    if (clientSpan != null && clientSpan.url() != null) {
+                        BinaryAnnotation httpURLAnnotation = new BinaryAnnotation();
+                        httpURLAnnotation.setKey("htp.url");
```

Comment:
Annotation key is wrong :)
Issue is that this is http specific - possibly if we had a generic 'uri' annotation as well that would be fallback?